app/logical/sources/strategies/fanbox.rb

```diff
@@ -0,0 +1,188 @@
+      end
+
+      def artist_name
+        artist_name_from_url || api_response["creatorId"] || artist_api_response["creatorId"]
+      end
+
+      def display_name
+        if api_response.present?
+          api_response["user"]["name"]
+        elsif artist_api_response.present?
+          artist_api_response["user"]["name"]
+        end
+      end
+
+      def other_names
+        [artist_name, display_name].compact.uniq
+      end
+
+      def tags
+        api_response["tags"].to_a.map { |tag| [tag, "https://fanbox.cc/tags/#{tag}"] }
+      end
+
+      def artist_commentary_title
+        api_response["title"]
+      end
+
+      def artist_commentary_desc
+        return if api_response.blank?
+        body = api_response["body"]
+        if body["text"].present?
+          body["text"]
+        elsif body["blocks"].present?
+          # Reference: https://official.fanbox.cc/posts/182757
+          # Commentary can get pretty complex, but unfortunately it's served in json format so it's a pain to parse it.
+          # I've left out parsing external embeds because each supported site has its own id mapped to the domain
+          commentary = body["blocks"].map do |node|
+            if node["type"] == "image"
+              body["imageMap"][node["imageId"]]["originalUrl"]
+            else
+              node["text"] || "\n"
+            end
+          end
+          commentary.join("\n")
+        end
+      end
+
+      def illust_id
+        urls.map { |url| url[PAGE_NEW, :illust_id] || url[IMAGE, :illust_id] || url[PAGE_OLD, :illust_id] || url[OTHER_IMAGES, :illust_id] }.compact.first
+      end
+
+      def artist_id_from_url
+        urls.map { |url| url[PAGE_OLD, :artist_id] || url[OTHER_IMAGES, :artist_id] }.compact.first
+      end
+
+      def artist_name_from_url
+        urls.map { |url| url[PROFILE_NEW, :artist_name] }.compact.first
+      end
+
+      def api_response
+        return {} if illust_id.blank?
+        resp = client.get("https://api.fanbox.cc/post.info?postId=#{illust_id}")
+        json_response = JSON.parse(resp)["body"]
+        if json_response["restrictedFor"] == 2 && json_response["body"].blank?
+          # Pixiv Fanbox login is protected by Google Recaptcha, so it's not possible for us to extract anything from them (save for the title).
+          # Other projects like PixivUtils ask the user to periodically extract cookies from the browser, but this is not feasible for Danbooru.
+          raise Sources::Error, "Age-restricted posts from Pixiv Fanbox are not supported."
+        end
+
+        json_response
+      rescue JSON::ParserError
+        {}
+      end
+
+      def artist_api_response
+        # Needed to fetch artist from cover pages
+        return {} if artist_id_from_url.blank?
+        resp = client.get("https://api.fanbox.cc/creator.get?userId=#{artist_id_from_url}")
+        JSON.parse(resp)["body"]
+      rescue JSON::ParserError
+        {}
+      end
```

Comment:
In what situation can parsing the json fail? If it's when a non-200 response is returned, it would be better to check for that case explicitly.

Reply:
During my mass tests there have been a few times where the result was garbled HTML or badly formatted japanese, which is why it wasn't reliable to parse the JSON without a rescue. Other than that, there's never any other errors as long as the origin header is passed.